x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/fulltext/FullTextFunction.java

```diff
@@ -462,7 +464,7 @@ public static String getNameFromFieldAttribute(FieldAttribute fieldAttribute) {
         return fieldName;
     }
 
-    public static FieldAttribute fieldAsFieldAttribute(Expression field) {
+    protected FieldAttribute fieldAsFieldAttribute(Expression field) {
         Expression fieldExpression = field;
         // Field may be converted to other data type (field_name :: data_type), so we need to check the original field
         if (fieldExpression instanceof AbstractConvertFunction convertFunction) {
```

Comment:
A somewhat lateral note: we could have an unpacking loop here, for the (more likely automated query generation) case of nested conversions, like: `field::INTEGER::LONG`.